### ISSUE TITLE
Cannot start workspaces on K8S azure loadbalancer can't bind to cluster ip

### ISSUE THREAD

**Issue by bs-matil** [eclipse-che/che#9914]
### Description
Hi all

AKS V1.9.2

I got a strange issue with che 6.3 deployed in k8s.  if I try to open a workspace the page starts to load but the workspace is never created

```
2018-06-01 13:21:24,803[main]             [INFO ] [o.a.c.http11.Http11NioProtocol 632]  - Starting ProtocolHandler ["http-nio-8080"]
2018-06-01 13:21:24,850[main]             [INFO ] [o.a.catalina.startup.Catalina 683]   - Server startup in 20890 ms
2018-06-01 13:22:52,273[nio-8080-exec-1]  [INFO ] [o.e.c.a.w.s.WorkspaceManager 461]    - Workspace 'test/wksp-63lu' with id 'workspacenee7lz0ophft9npc' created by user 'test'
2018-06-01 13:22:59,345[nio-8080-exec-9]  [INFO ] [o.e.c.a.w.s.WorkspaceRuntimes 228]   - Starting workspace 'test/wksp-63lu' with id 'workspacenee7lz0ophft9npc' by user 'test'
2018-06-01 13:28:00,300[aceSharedPool-0]  [WARN ] [.i.k.KubernetesInternalRuntime 162]  - Failed to start Kubernetes runtime of workspace workspacenee7lz0ophft9npc. Cause: Waiting for ingress 'ingress3uozxvtv' reached timeout
2018-06-01 13:28:02,466[aceSharedPool-0]  [INFO ] [o.e.c.a.w.s.WorkspaceRuntimes 284]   - Workspace 'test:wksp-63lu' with id 'workspacenee7lz0ophft9npc' start failed
```

![image pasted at 2018-6-1 16-38](https://user-images.githubusercontent.com/5337267/40846968-b3fed942-65c3-11e8-9125-0d919fd8cd5f.png)

**Comment by bsukramlitt** [comment]
I Also notices `Failed to start Kubernetes runtime of workspace workspacenee7lz0ophft9npc. Cause: Waiting for ingress` in the logs

**Comment by bsukramlitt** [comment]
Turns out its a combination of a zero impact rated bug of aks 1.9.x < 1.9.6 which is caused by the way you check ingress to be ready 
`  // wait for LB ip
    List<Ingress> readyIngresses = new ArrayList<>();
    for (Ingress ingress : createdIngresses) {
      Ingress actualIngress =
          namespace
              .ingresses()
              .wait(
                  ingress.getMetadata().getName(),
                  ingressStartTimeout,
                  p -> (!p.getStatus().getLoadBalancer().getIngress().isEmpty()));
      readyIngresses.add(actualIngress);
    }`
This line causes a failure in combination with an bug in azure which on its own causes loadbalancer binding failures on ClusterIP services.
`p.getStatus().getLoadBalancer().getIngress().isEmpty())`

see https://github.com/kubernetes/kubernetes/issues/60528 (azure cloud provider tries to get loadbalancer for services that do not have Type=LoadBalancer #60528)

In the end. Its fixed by updateingto AKS 1.9.6.

unfortunately I have now different issues. But will open a new ticket if I know more.

**Comment by bsukramlitt** [comment]
@eivantsov please close
